core/src/main/java/io/parsingdata/metal/Shorthand.java

```diff
@@ -90,22 +91,22 @@
 public final class Shorthand {
 
     public static final Token EMPTY = def(EMPTY_NAME, 0L);
-    public static final ValueExpression SELF = new Self();
-    public static final ValueExpression CURRENT_OFFSET = new CurrentOffset();
-    public static final ValueExpression CURRENT_ITERATION = new CurrentIteration(con(0));
+    public static final SingleValueExpression SELF = new Self();
+    public static final SingleValueExpression CURRENT_OFFSET = new CurrentOffset();
+    public static final SingleValueExpression CURRENT_ITERATION = new CurrentIteration(con(0));
     public static final Expression TRUE = new True();
 
     private Shorthand() {}
 
-    public static Token def(final String name, final ValueExpression size, final Expression predicate, final Encoding encoding) { return post(def(name, size, encoding), predicate); }
-    public static Token def(final String name, final ValueExpression size, final Expression predicate) { return def(name, size, predicate, null); }
-    public static Token def(final String name, final ValueExpression size, final Encoding encoding) { return new Def(name, size, encoding); }
-    public static Token def(final String name, final ValueExpression size) { return def(name, size, (Encoding)null); }
+    public static Token def(final String name, final SingleValueExpression size, final Expression predicate, final Encoding encoding) { return post(def(name, size, encoding), predicate); }
```

Comment:
Now I see this, and I can't recall why: what is the reason we do not allow a list of sizes? Apart from that it could be written down differently (and better?), and that there are not really use cases for it then. Is it an arbitrary choice? Just curious, I think it makes sense of course.

Reply:
I think we just decided that if you want a list of `def`s, you just need to use `seq` (with `repn` as dynamic alternative).